devservices/s3/src/main/java/io/quarkus/amazon/devservices/s3/S3DevServicesProcessor.java

```diff
@@ -0,0 +1,82 @@
+package io.quarkus.amazon.devservices.s3;
+
+import java.util.Objects;
+import java.util.Set;
+
+import org.testcontainers.containers.localstack.LocalStackContainer;
+import org.testcontainers.containers.localstack.LocalStackContainer.Service;
+
+import io.quarkus.amazon.common.deployment.spi.AbstractDevServicesLocalStackProcessor;
+import io.quarkus.amazon.common.deployment.spi.DevServicesLocalStackProviderBuildItem;
+import io.quarkus.amazon.common.runtime.DevServicesBuildTimeConfig;
+import io.quarkus.amazon.s3.runtime.S3BuildTimeConfig;
+import io.quarkus.amazon.s3.runtime.S3DevServicesBuildTimeConfig;
+import io.quarkus.deployment.annotations.BuildStep;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.CreateBucketRequest;
+
+public class S3DevServicesProcessor extends AbstractDevServicesLocalStackProcessor {
+
+    @BuildStep
+    DevServicesLocalStackProviderBuildItem setupS3(S3BuildTimeConfig clientBuildTimeConfig) {
+        return this.setup(Service.S3, clientBuildTimeConfig.devservices);
+    }
+
+    @Override
+    protected void prepareLocalStack(DevServicesBuildTimeConfig clientBuildTimeConfig, LocalStackContainer localstack) {
+        createBuckets(localstack, getConfiguration((S3DevServicesBuildTimeConfig) clientBuildTimeConfig));
+    }
+
+    public void createBuckets(LocalStackContainer localstack, S3DevServiceCfg configuration) {
+        try (S3Client client = S3Client.builder()
```

Comment:
Just to be sure: we are using the v2 API here, right?

Reply:
Yes